src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/stdx/type_traits.hpp

```diff
@@ -0,0 +1,428 @@
+#pragma once
+
+#include <type_traits>
+
+#include <bsoncxx/config/prelude.hpp>
+
+namespace bsoncxx {
+
+BSONCXX_INLINE_NAMESPACE_BEGIN
+namespace _traits {
```

Comment:
To clarify/elaborate, this recommendation is to ensure _all_ the polyfill type trait declarations we use are declared in the `details` namespace to document+ensure we do not become beholden to maintaining the stability of our type trait implementations as part of the public API/ABI (they are just a means to an end; we do not want to be a type traits polyfill library provider).

Ideally this should extend to other polyfill entities as well, but those (optional and string_view in particular) are too baked into the API/ABI atm to allow for this treatment.

Reply:
I simplified/flattened the namespace structure further. I wanted to avoid heavy nesting due to diagnostics sizes and name resolution issues. Having a flat `detail` is sufficient, since we control the namespace and won't collide with ourself.